test/e2e/network_test.go

```diff
@@ -574,6 +575,30 @@ func (tc *testContext) getPodEvents(name string) ([]v1.Event, error) {
 	return podEvents, nil
 }
 
+// createLinuxCurlerJob creates a linux job to curl a specific endpoint. curl must be present in the container image.
+func (testCtx *testContext) createLinuxCurlerJob(jobSuffix, endpoint, portNum, portName string) (*batch.Job, error) {
+	if portNum != "" {
+		endpoint = fmt.Sprintf("%s:%s", endpoint, portNum)
+	}
+	if portName != "" {
+		endpoint = fmt.Sprintf("%s/%s", endpoint, portName)
+	}
+	// curl endpoint every 5 seconds. If the webserver is not accessible for a few minutes, the pod exits with an error.
+	command := []string{
+		"bash",
+		"-c",
+		" while true;" +
+			" do curl " + endpoint + ";" +
+			" if [ $? != 0 ]; then" +
+			" sleep 60;" +
+			" curl " + endpoint + " || exit 1;" +
```

Comment:
Why do we need to do curl twice here? I think this can be collapsed into one.

Reply:
Not sure, let me try to rework it

Reply:
This code is pretty convoluted and can be simplified 😄

Reply:
@aravindhp After a bit I was able to see the purpose of it. I added some docs around it

Reply:
I still don't see why it needs to be this way. If we are trying to avoid flakes, we just retry the curl command N number of times and exit with an error.